README.md
![alt XOOPS CMS](http://xoops.org/images/logoXoops4GithubRepository.png)
## RandomQuote module for [XOOPS CMS 2.5.8+](https://xoops.org)
[![XOOPS CMS Module](https://img.shields.io/badge/XOOPS%20CMS-Module-blue.svg)](http://xoops.org)
[![Software License](https://img.shields.io/badge/license-GPL-brightgreen.svg?style=flat)](LICENSE)

[![Scrutinizer Code Quality](https://img.shields.io/scrutinizer/g/mambax7/randomquote.svg?style=flat)](https://scrutinizer-ci.com/g/mambax7/randomquote/?branch=master)
[![Codacy Badge](https://api.codacy.com/project/badge/grade/2d27c0023ee54f0b9ba2b5d17a68b2a5)](https://www.codacy.com/app/mambax7/randomquote)
[![Code Climate](https://img.shields.io/codeclimate/github/mambax7/randomquote.svg?style=flat)](https://codeclimate.com/github/mambax7/randomquote)
[![SensioLabsInsight](https://insight.sensiolabs.com/projects/bb99aca9-fd51-49e9-98ae-6fd9f51bc7a9/mini.png)](https://insight.sensiolabs.com/projects/bb99aca9-fd51-49e9-98ae-6fd9f51bc7a9)
[![Latest Pre-Release](https://img.shields.io/github/tag/XoopsModules25x/randomquote.svg?style=flat)](https://github.com/XoopsModules25x/randomquote/tags/)
[![Latest Version](https://img.shields.io/github/release/XoopsModules25x/randomquote.svg?style=flat)](https://github.com/XoopsModules25x/randomquote/releases/)

RandomQuote module for [XOOPS CMS](http://xoops.org) adds a block that show a random quote. The quotes are added/edited/deleted in the Admin section.

[![Tutorial Available](http://xoops.org/images/tutorial-available-blue.svg)](https://www.gitbook.com/book/xoops/randomquote-tutorial/) Tutorial: see [GitBook](https://www.gitbook.com/book/xoops/randomquote-tutorial/).
To contribute to the Tutorial, [fork it on GitHub](https://github.com/XoopsDocs/randomquote-tutorial)

[![Translations on Transifex](http://xoops.org/images/translations-transifex-blue.svg)](https://www.transifex.com/xoops) 

Please visit us on  [http://xoops.org](http://xoops.org)

The upcoming "next generation" version of XOOPS CMS is being crafted on GitHub at: https://github.com/XOOPS


docs/changelog.txt
<h5>2.12 RC 1 [2017-6-23]</h5>
<hr>
- replaced "< br />" with "< br >" (mamba)
- added data sample (mamba)
- replaced XoopsRequest with Request (mamba)
- renamed Utilities to Utility (mamba)
- started conversion to XMF (mamba)
    - updated addInfoBoxLine entries (mamba)
    - displayNavigation(basename(__FILE__))  (mamba)
    - changed to ->displayButton('left') (mamba)


<h5>2.11 Final [2017-5-24]</h5>
<hr>
- cosmetics

<h5>2.11 Beta 1 [2016-5-16]</h5>
<hr>
- added breadcrumb display to top of index page
- added display of number of offline/online/waiting quotes in Admin index
- added significant security improvements for all vars passed from forms
- added module search capability
- added support for XOOPS Tag module
- added date quote was added to database
- changed to allow HTML in quote entry (XoopsRequest::getText) in admin
- changed install/update/unistall scripts
- changed hard coded language strings from templates to language files
- changed to improved index.tpl rendering
- changed to improved PSRx naming nomenclature
- changed phpDocumentor copyright, license, & link elements
- changed to clean up phpdocumentor comments
- fixed typo in block text length calculation
- fixed bug where 'item_tag' wasn't correctly assigned to template in /class/quotes.php
- fixed bug in install script PHP version checking
- fixed bug in install script XOOPS version checking
- fixed bug where language file wasn't being loaded in install script
- fixed typo in ./docs/licencs file name to license.txt
- general code cleanup
- removed deprecated cleanVarsRandomquote, replaced with XoopsRequest
- removed '/' from void HTML elements (&lt;br&gt;, &lt;img&gt;, etc)
- removed need for ./include/config.php
- removed need for ./include/action.module.php
- removed @version from phpDocumentor form page-level blocks
- removed unused admin templates
- fixed PHP version check (mamba)
- fixed uninstall (mamba)

<h5>2.10 Beta 1 [2016-04-16]</h5>
<hr>
- Updates for XOOPS 2.5.8, PHP 7 (mamba)
- Set min PHP to 5.5

<h5>2.00 [2013-06-27]</h5>
<hr>
 - original release RandomQuote 2.0 Alpha 1 (mamba)
   This module is developed from scratch using TDMCreate, but it's inspired by
   the original code from RandomQuote


xoops_version.php
<?php
/*
 You may not change or alter any portion of this comment or credits of
 supporting developers from this source code or any supporting source code
 which is considered copyrighted (c) material of the original comment or credit
 authors.

 This program is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
/**
 * Module: RandomQuote
 *
 * @category        Module
 * @package         randomquote
 * @author          XOOPS Module Development Team
 * @author          Mamba
 * @copyright       {@link http://xoops.org 2001-2016 XOOPS Project}
 * @license         {@link http://www.fsf.org/copyleft/gpl.html GNU public license}
 * @link            http://xoops.org XOOPS
 * @since           2.00
 */

// defined('XOOPS_ROOT_PATH') || exit('XOOPS root path not defined');

$moduleDirName = basename(__DIR__);
xoops_load('XoopsLists');
xoops_load('xoopseditorhandler');
$editorHandler = XoopsEditorHandler::getInstance();

$modversion = array(
    'version'             => '2.12',
    'module_status'       => 'RC 1',
    'release_date'        => '2017/6/23',
    'name'                => _MI_RANDOMQUOTE_ADMIN_NAME,
    'description'         => _MI_RANDOMQUOTE_ADMIN_DESC,
    'official'            => 0,
    //1 if supported by XOOPS CORE Dev Team, 0 otherwise
    'author'              => 'XOOPS Module Development Team',
    //          'author_mail' => 'webmaster@xoops.org',
    'author_website_url'  => 'http://xoops.org',
    'author_website_name' => 'XOOPS Project',
    'credits'             => 'XOOPS Module Development Team',
    'license'             => 'GPL 2.0 or later',
    'license_url'         => 'www.gnu.org/licenses/gpl-2.0.html',
    'help'                => 'page=help',
    'helpsection'         => array(
        array(
            'name' => _MI_RANDOMQUOTE_ADMIN_HELP,
            'link' => 'page=help'
        ),
        array(
            'name' => _MI_RANDOMQUOTE_ADMIN_HELP_FEATURES,
            'link' => 'page=module_index'
        )
    ),

    //                  'manual' => 'install.txt',
    //             'manual_file' => XOOPS_URL . "/modules/{$moduleDirName}/docs/link to manual file",
    'min_php'             => '5.5',
    'min_xoops'           => '2.5.8',
    'min_admin'           => '1.2',
    'min_db'              => array('mysql' => '5.5'),
    'image'               => 'assets/images/logoModule.png',
    'dirname'             => $moduleDirName,

    'modicons16'          => 'assets/images/icons/16',
    'modicons32'          => 'assets/images/icons/32',

    // About
    //            'release_info' => 'module_release_info',
    //            'release_file' => XOOPS_URL . "/modules/{$moduleDirName}/docs/module_release_info file",
    //           'demo_site_url' => "http://xoops.org",
    //          'demo_site_name' => 'XOOPS Demo Site',
    //          'forum_site_url' => "http://xoops.org",
    //         'forum_site_name' => 'XOOPS Project',
    'module_website_url'  => 'http://xoops.org/modules/newbb',
    'module_website_name' => 'XOOPS Project',

    // Admin things
    'hasAdmin'            => 1,
    // Admin system menu
    'system_menu'         => 1,
    'hasMain'             => 1,
    'adminindex'          => 'admin/index.php',
    'adminmenu'           => 'admin/menu.php',
    // Mysql file
    'sqlfile'             => array('mysql' => 'sql/mysql.sql'),
    // Tables
    'tables'              => array("{$moduleDirName}_quotes"),

    // Scripts to run upon installation or update
    'onInstall'           => 'include/oninstall.php',
    'onUpdate'            => 'include/onupdate.php',
    'onUninstall'         => 'include/onuninstall.php',

    // Search
    'hasSearch'           => 1,
    'search'              => array(
        'file' => 'include/search.inc.php',
        'func' => $moduleDirName . '_search'
    ),
    // Templates
    'templates'           => array(
        array(
            'file'        => $moduleDirName . '_index.tpl',
            'description' => 'randomquote index page'
        ),
        /*
                                            array('file' => $moduleDirName . '_quotes.tpl',
                                           'description' => 'randomquote quotes page'),
        */
        array(
            'file'        => $moduleDirName . '_header.tpl',
            'description' => 'randomquote header page'
        ),

        array(
            'file'        => $moduleDirName . '_footer.tpl',
            'description' => 'randomquote footer page'
        )
    ),
    //Blocks
    'blocks'              => array(
        array(
            'file'        => 'views.php',
            'name'        => _MI_RANDOMQUOTE_QUOTES_BLOCK_RECENT,
            'description' => _MI_RANDOMQUOTE_QUOTES_BLOCK_RECENT_DESC,
            'show_func'   => 'showRandomquoteBlockViews',
            'edit_func'   => 'editRandomquoteBlockViews',
            'options'     => 'recent|5|25|0',
            'template'    => $moduleDirName . '_quotes_block_recent.tpl'
        ),

        array(
            'file'        => 'views.php',
            'name'        => _MI_RANDOMQUOTE_QUOTES_BLOCK_DAY,
            'description' => _MI_RANDOMQUOTE_QUOTES_BLOCK_DAY_DESC,
            'show_func'   => 'showRandomquoteBlockViews',
            'edit_func'   => 'editRandomquoteBlockViews',
            'options'     => 'day|5|25|0',
            'template'    => $moduleDirName . '_quotes_block_day.tpl'
        ),

        array(
            'file'        => 'views.php',
            'name'        => _MI_RANDOMQUOTE_QUOTES_BLOCK_RANDOM,
            'description' => _MI_RANDOMQUOTE_QUOTES_BLOCK_RANDOM_DESC,
            'show_func'   => 'showRandomquoteBlockViews',
            'edit_func'   => 'editRandomquoteBlockViews',
            'options'     => 'random|5|25|0',
            'template'    => $moduleDirName . '_quotes_block_random.tpl'
        )
    ),

    // Config
    'config'              => array(
        array(
            'name'        => 'randomquote_editor',
            'title'       => '_MI_RANDOMQUOTE_EDITOR',
            'description' => '_MI_RANDOMQUOTE_EDITOR_DESC',
            'formtype'    => 'select',
            'valuetype'   => 'text',
            'default'     => 'dhtml',
            'options'     => array_flip($editorHandler->getList())
        ),

        array(
            'name'        => 'keywords',
            'title'       => '_MI_RANDOMQUOTE_KEYWORDS',
            'description' => '_MI_RANDOMQUOTE_KEYWORDS_DESC',
            'formtype'    => 'textbox',
            'valuetype'   => 'text',
            'default'     => ''
        ),

        array(
            'name'        => 'advertise',
            'title'       => '_MI_RANDOMQUOTE_ADVERTISE',
            'description' => '_MI_RANDOMQUOTE_ADVERTISE_DESC',
            'formtype'    => 'textarea',
            'valuetype'   => 'text',
            'default'     => ''
        ),
        /*
                                                    array('name' => 'social_bookmarks',
                                                         'title' => '_MI_RANDOMQUOTE_SOCIAL_BOOKMARKS',
                                                   'description' => '_MI_RANDOMQUOTE_SOCIAL_BOOKMARKS_DESC',
                                                      'formtype' => 'yesno',
                                                     'valuetype' => 'int',
                                                       'default' => 0),

                                                    array('name' => 'fbcomments',
                                                         'title' => '_MI_RANDOMQUOTE_FBCOMMENTS',
                                                   'description' => '_MI_RANDOMQUOTE_FBCOMMENTS_DESC',
                                                      'formtype' => 'yesno',
                                                     'valuetype' => 'int',
                                                       'default' => 0)
                */
    )
);
